paasta_tools/cli/cmds/secret.py

```diff
@@ -401,7 +402,19 @@ def paasta_secret(args):
             sys.exit(1)
 
         kube_client = KubeClient(config_file=KUBE_CONFIG_USER_PATH, context=clusters[0])
-        print(get_kubernetes_secret(kube_client, service, args.secret_name))
+
+        secret_to_k8s_mapping = get_namespace_for_secret(
+            service, clusters[0], args.secret_name, args.yelpsoa_config_root
+        )
+        if secret_to_k8s_mapping:
+            # assuming that if multiple namespaces, they should be the same
```

Comment:
My only small nit is that if one of the namespaces is in some experimental namespace or something, it may have an out of date or incorrect value. I don't think there's any change to be made here, just mentioning it because I tend to think of pathological things like this :)

Reply:
Hmm the only other thing I can think of that we can do here is to create a priority_list = ["paasta", "tron", ..] to first check if these predefined non-experimental values are present in the secret_to_k8s_mapping and use them, otherwise fallback on .pop(). Do you think that would be useful in this case? If so, should it contain the set of values as they are now here: https://sourcegraph.yelpcorp.com/Yelp/paasta/-/blob/paasta_tools/utils.py?L140:1 ?

Reply:
Yeah that's a great idea. 🚢